src/renderers/dom/fiber/ReactDOMFiberEntry.js

```diff
@@ -531,6 +531,23 @@ function renderSubtreeIntoContainer(
   );
 
   if (__DEV__) {
+    let warned;
+    if (!warned && container._reactRootContainer) {
+      const firstChild = container.firstChild;
+      const hostInstance =
+        DOMRenderer.findHostInstance(container._reactRootContainer.current);
+      if (hostInstance) {
+        warned = true;
+        warning(
+          hostInstance.parentNode === container,
+          'render(...): It looks like the content of this container may have ' +
+          'been updated or cleared outside of React. ' +
+          'This can cause errors or failed updates to the container. ' +
+          'Please call `ReactDOM.render` with your new content.',
```

Comment:
How about:

> render(...): It looks like the React-rendered content of this container was removed without using React. This is not supported and will cause errors. Instead, call ReactDOM.unmountComponentAtNode to empty a container.

since it's not likely that it is simple for people to switch to using React to render things – we just want them to use React to clean things up properly.